includes/utils.php

```diff
@@ -245,21 +245,25 @@ function set_meta( $post_id, $meta ) {
  * @return array
  */
 function available_pull_post_types( $connection, $type ) {
-	$post_types        = array();
-	$local_post_types  = array();
-	$remote_post_types = $connection->get_post_types();
-
-	if ( ! empty( $remote_post_types ) && ! is_wp_error( $remote_post_types ) ) {
-		$local_post_types     = array_diff_key( get_post_types( [ 'public' => true ], 'objects' ), array_flip( [ 'attachment', 'dt_ext_connection', 'dt_subscription' ] ) );
-		$available_post_types = array_intersect_key( $remote_post_types, $local_post_types );
-
-		if ( ! empty( $available_post_types ) ) {
-			foreach ( $available_post_types as $post_type ) {
-				$post_types[] = array(
-					'name' => 'external' === $type ? $post_type['name'] : $post_type->label,
-					'slug' => 'external' === $type ? $post_type['slug'] : $post_type->name,
-				);
-			}
+	$post_types               = array();
+	$local_post_types         = array();
+	$remote_post_types        = $connection->get_post_types();
+	$distributable_post_types = distributable_post_types();
+
+	// Return empty array, if the source site is not distributing any post type.
```

Comment:
Something to be aware of with the change to the early `return` pattern.

In the current version of the plugin, an empty array would still be passed to `dt_available_pull_post_types` for developers to modify as they wish. With this change, the empty array is no longer passed through the filter.

I think it will be fine though.